### DISCUSSION TITLE
gulpfile.ts initial compilation time

### DISCUSSION THREAD

**Discussion by chasset** [gulpjs/gulp#2448] [Help] [answered]
Hello,

I prefer writing my tasks in typescript. So I use a folder named "gulpfile.ts" to organize all my scripts dedicated to gulp. After installing ts-node, "gulp" on the command line compiles all the scripts and launch the tasks perfectly.

But the process to compile each time is slow. Now, I would like to speed the initial compilation using the incremental option of the compiler. 

But it seems that it doesn’t care of the "tsconfig.json" file, using may be its own parameters.

Any idea to make it compile faster?

Best regards,

Pierre-Olivier

**Comment by phated** [comment, marked as answer]
I'm not too sure what `ts-node` would be doing behind the scenes, but I was pretty sure it uses the `tsconfig` file (I'm think we test that in our `interpret` module). However, yesterday, we added support for a new library called [Sucrase](https://www.npmjs.com/package/sucrase) that is supposed to allow for very fast builds.

You can try it out by removing `ts-node` from your project and installing `sucrase`.

**Reply by chasset** [reply]
It works. Many thanks !

**Comment by chasset** [comment]
Sucrase reduces compilation time, but I have still 9 secondes to wait.
Is there a way to use an incremental compilation for typescript in combination with sucrase?
